dhis-2/dhis-services/dhis-service-validation/src/main/resources/org/hisp/dhis/validation/hibernate/ValidationRule.hbm.xml

```diff
@@ -34,10 +34,10 @@
     <property name="operator" column="operator" type="org.hisp.dhis.expression.OperatorUserType" not-null="true" />
 
     <many-to-one name="leftSide" column="leftexpressionid" class="org.hisp.dhis.expression.Expression"
-                 cascade="all" foreign-key="fk_validationrule_leftexpressionid"/>
+                 cascade="all" foreign-key="fk_validationrule_leftexpressionid" not-null="true"/>

```

Comment:
Although it might not have been missing for a long time, we should have a flyway script to make sure we update the data before we try to make it not-null true again. I think either an empty "" or "0" expression should be correct. You can maybe have a chat with @jimgrace to see what makes sense in the case the expression is null.